### PR TITLE
[navigation-next] fix: redirect to standard index pattern applications while nav group is enabled

### PR DESCRIPTION
**Pull request by SuZhou-Joe**
### Description

When nav group is enabled and users want to visit `Dashboards` page if no index pattern is configured in their OSD, OSD will direct user to the index pattern page, which is a sub app under management.

This PR is mainly to solve this problem by redirecting users to the right application.

- Address the issue that in-app navigation is not hide when nav group is enabled.
- Address the issue that the app will still navigate to legacy index pattern management oage.
- Address the scss code concern from @AMoo-Miki 

### Before the fix
https://github.com/user-attachments/assets/34df4f15-9d2d-44ea-8875-bb51db117751

### After the fix
https://github.com/user-attachments/assets/a902d738-25a3-4a7a-a685-882531eb6a3c

## Changelog

- feat: [navigation-next] fix: redirect to standard index pattern applications while nav group is enabled

### Check List

- [x] All tests pass
  - [x] `yarn test:jest`
  - [x] `yarn test:jest_integration`
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [x] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/core/public/chrome/nav_group/nav_group_service.ts [hunk 1]
```diff
@@ -213,7 +213,7 @@ export class ChromeNavGroupService {
     };
 
     // erase current nav group when switch app don't belongs to any nav group
-    application.currentAppId$.subscribe((appId) => {
+    application.currentAppId$.pipe(debounceTime(500)).subscribe((appId) => {
```
Comment (virajsanghvi) [inline src/core/public/chrome/nav_group/nav_group_service.ts:216]: Why do you need to debounce?

**Review by virajsanghvi — COMMENTED** [review]
Can you update after video (think its the same as before)?

File: src/core/public/chrome/nav_group/nav_group_service.ts (hunk 1 above)
Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts:220]: Sure, added some comment.

**Comment by SuZhou-Joe** [comment]
> Can you update after video (think its the same as before)?

Thanks for catching this, updated with correct video.

File: src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss [hunk 2]
```diff
@@ -2,8 +2,8 @@
   border: none !important;
 
   .nav-link-item {
-    padding: $ouiSize / 4 $ouiSize;
-    border-radius: $ouiSize;
+    padding: calc($euiSize / 4) $euiSize;
+    border-radius: $euiSize;
```
Comment (ruanyl) [inline src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:6]: Are these changes expected? Seems unrelated

Comment (SuZhou-Joe) [reply src/core/public/chrome/ui/header/collapsible_nav_group_enabled.scss:6]: It is expected, as Miki told me to do so.

File: src/core/public/chrome/nav_group/nav_group_service.ts (hunk 1 above)
Comment (ruanyl) [reply src/core/public/chrome/nav_group/nav_group_service.ts:220]: I feel it would be better to have the debounceTime here, so that the consumer that subscribe to `currentNavGroupSorted$` to update UI wouldn't cause the UI flashing when the value changed so frequently.

```typescript
    const currentNavGroupSorted$ = combineLatest([
      this.getSortedNavGroupsMap$(),
      this.currentNavGroup$,
    ])
      .pipe(takeUntil(this.stop$))
      .pipe(debounceTime(500))
      .pipe(
        map(([navGroupsMapSorted, currentNavGroup]) => {
          if (currentNavGroup) {
            return navGroupsMapSorted[currentNavGroup.id];
          }
        })
      );

```

Comment (ruanyl) [reply src/core/public/chrome/nav_group/nav_group_service.ts:220]: Btw, what's the consequence if no `debounceTime(500)`? Anything UI broken?

Comment (SuZhou-Joe) [reply src/core/public/chrome/nav_group/nav_group_service.ts]: I just removed the reset logic because:
When redirecting to discover, the appId will get changed twice, first is discover the latter is data-explorer. The reset logic will always reset the currentNavGroup to undefined because data-explorer is a hidden application that won't be registered in left navigation in the left navigation.

Comment (Hailong-am) [reply src/core/public/chrome/nav_group/nav_group_service.ts]: > The effect is that only when user click the `back` button in the new navigation will we reset the currentNavGroup to undefined.

make sense
